src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -90,17 +99,35 @@
 import org.apache.cassandra.schema.ColumnMetadata;
 import org.apache.cassandra.schema.IndexMetadata;
 import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.service.ClientWarn;
 import org.apache.cassandra.service.StorageService;
 import org.apache.cassandra.utils.Pair;
 import org.apache.cassandra.utils.concurrent.OpOrder;
 
+import static org.apache.cassandra.index.sai.disk.v1.IndexWriterConfig.MAX_TOP_K;
+
 public class StorageAttachedIndex implements Index
 {
     public static final String NAME = "sai";
+    
+    public static final String VECTOR_USAGE_WARNING = "SAI ANN indexes on vector columns are experimental and are not recommended for production use.\n" +
```

Comment:
I have just realized that the documentation for SAI committed by CASSANDRA-18231 a month ago contains documentation about vector search. We should verify that the doc matches this implementation, and probably add the list of query limitations to it, probably [here](https://github.com/apache/cassandra/blob/trunk/doc/modules/cassandra/partials/vector-search/query-vector-data.adoc).

Reply:
That documentation seems to contain a bunch of unreachable documents, if I'm understanding the links correctly. I'd say it has been accidentally committed. So probably it's better to ignore that by now and have a separate ticket for the doc.

Reply:
I agree, let's deal with that separately. I have dropped a comment on CASSANDRA-18231.